File: f77/build10/def_data.f
C
      Subroutine Default_Data
C
C     This subroutine calls the routines responsible for setting
C     default data values. It also sets default system values for this
C     simulation.
C
C     Inputs:
C       None.
C
C     Outputs:
C       Dt      - Integration time step [sec]
C       DtMax   - Maximum allowable Dt [sec]
C       DtMin   - Minimum allowable Dt [sec]
C       DtPrint - Print data logging interval [sec]
C       Time0   - Initial time [sec]
C       Tstop   - Simulation termination time [sec]
C
C     Internal variables and constants:
C       PI     - Ratio of a circle's circumference to its diameter [Real]
C       RDTODG - Radians to degrees conversion factor [deg/rad]
C       SMALL  - Arbitrarily small number [Real]
C
C     Declare global common
C
      Include '../driver/global.inc'
      Include '../driver/sysvars.inc'
      Equivalence (Real_Array (50), PI)
      Equivalence (Real_Array (51), RDTODG)
      Equivalence (Real_Array (52), SMALL)
C
C     Set model independent data and reset time
C
      Dt      =  0.005e0
      DtMax   =  0.005e0
      DtMin   =  0.001e0
      DtPrint =  0.01e0
      Time0   =  0.0e0
      Tstop   = 10.0e0
C
C       Commonly used constants
C
      PI     = 4.0e0*Atan (1.0e0)
      RDTODG = 180.0e0/PI
      SMALL  = 1.0e-6
C
C     Call routines to set default data
C
      Call Actuator_Data
      Call Aerodynamics_Data
      Call Mass_Data
      Call Kinematics_Data
      Call Gyro_Data
      Call Target_Data
      Call Seeker_Data
      Call Flight_Computer_Data
C
      Return
      End

File: gcc/build10/def_data.c
#include "def_data.h"
#include "../driver/global.h"
#include "../driver/sysvars.h"
#include <math.h>
#include "actuator.h"
#include "actuator.h"
#include "aero.h"
#include "mass.h"
#include "kmtc.h"
#include "gyro.h"
#include "target.h"
#include "seeker.h"
#include "fcomp.h"

#define pi     real_array[50].r
#define rdtodg real_array[51].r
#define small  real_array[52].r

void default_data() {
  // This subroutine calls the routines responsible for setting
  // default data values. It also sets default system values for this
  // simulation.
  //
  // Inputs:
  //   None.
  //
  // Outputs:
  //   Dt      - Integration time step [sec]
  //   DtMax   - Maximum allowable Dt [sec]
  //   DtMin   - Minimum allowable Dt [sec]
  //   DtPrint - Print data logging interval [sec]
  //   Time0   - Initial time [sec]
  //   Tstop   - Simulation termination time [sec]
  // Set model independent data and reset time
  dt = 0.005;
  dtmax = 0.005;
  dtmin = 0.001;
  dtprint = 0.01;
  time0 = 0.0;
  tstop = 10.0;
  // Commonly used constants
  pi = 4.0*atan(1.0);
  rdtodg = 180.0/pi;
  small = 0.000001;
  // Call routines to set default data
  actuator_data();
  aerodynamics_data();
  mass_data();
  kinematics_data();
  gyro_data();
  target_data();
  seeker_data();
  flight_computer_data();
}
